source/extensions/filters/http/transformation/body_header_transformer.cc

```diff
@@ -62,6 +67,94 @@ void BodyHeaderTransformer::transform(
   header_map.setContentLength(body.length());
 }
 
+void BodyHeaderTransformer::parse_headers(
+  const Http::RequestOrResponseHeaderMap &header_map,
+  std::map<std::string, absl::string_view> &headers,
+  std::map<std::string, std::vector<absl::string_view>> &multi_value_headers) const {
+    if (add_request_metadata_) {
+      header_map.iterate(
+        [&headers, &multi_value_headers](const Http::HeaderEntry& header) -> Http::HeaderMap::Iterate {
+            auto header_key = std::string(header.key().getStringView());
+            auto header_value = header.value().getStringView();
+            if (headers[header_key].empty()) {
+                headers[header_key] = header_value;
+            } else {
+                // If there are more than one headers with the same key, use the last one
+                auto existing_header_value = headers[header_key];
+                headers[header_key] = header_value;
+ 
+                // Then, ensure all existing values are in the multiValueHeaders vector
+                if (multi_value_headers[header_key].empty()) {
+                    multi_value_headers[header_key].push_back(existing_header_value);
```

Comment:
why do we use push_back here and emplace_back below? Is it situational, or is one more correct than the other?

Reply:
A useful explanation for the difference b/w the two is here: https://www.quora.com/In-what-scenarios-should-I-use-emplace_back-instead-of-push_back-in-C%2B%2B/answer/Kevin-Mather?ch=10&share=46e9e37c&srid=CWXc

It seems like emplace_back is faster than push_back in some situations, so I'm choosing to use that in each location.